src/Services/ReferenceGenerator.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace App\Services;
+
+use App\Repository\UserRepository;
+use App\Entity\User;
+use DateTime;
+
+class ReferenceGenerator
+{
+    private UserRepository $userRepository;
+
+    public function __construct(UserRepository $userRepository)
+    {
+        $this->userRepository = $userRepository;
+    }
+
+    public function generateReference(): string
+    {
+        $date = new DateTime();
+        $user = $this->userRepository->findByYear();
```

Comment:
$lastUser plutot

et if ($lastUser === null) {
$number = 1;
} else {
    $number = $this->getNumber($user);
}

et ça serait interessant d'avoir une méthode getNewReference($number) qui ferait juste la partie concatenation
```php
public function getNewReference(int $number)
{
    return 'CER_' . str_pad((string) $number+1, 4, '0', STR_PAD_LEFT) . '_' .
}

``` 
le +1 se ferait donc getNewReference et plus dans getNumber qui n'aurait comme but que de renvoyer un number d'un user en particulier